### PR TITLE
=sbt Bump reactive-stream-version to 1.0.4

### PR DESCRIPTION
**Pull request by He-Pin**
Bump Reactive Stream version to 1.0.4
https://github.com/reactive-streams/reactive-streams-jvm/blob/master/RELEASE-NOTES.md

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -59,20 +59,11 @@ object Dependencies {
   }
 
   val scalaTestVersion = Def.setting {
-    if (scalaVersion.value.startsWith("3.")) {
-      "3.2.9"
-    } else {
-      "3.1.4"
```
Comment (raboof) [inline project/Dependencies.scala:65]: Earlier (https://github.com/akka/akka/issues/29368#issuecomment-679843504) we decided not to update scalatest yet.

**Comment by He-Pin** [comment]
https://mvnrepository.com/artifact/org.scalatestplus/testng-7-5
testng is only for scalatest 3.2.x

**Comment by johanandren** [comment]
I don't think this is worth the effort, the only change that matters between 1.0.3 and 1.0.4 is this one https://github.com/reactive-streams/reactive-streams-jvm/commit/b008867c90b30f99b07999c0f5546d2006f87cc6 the rest is relicensing under MiT, rule wording, and one spec error message fix.

**Comment by johanandren** [comment]
And that change is only to cover a corner case with what method names are in the stack trace if I understand it correctly, which is not a problem in Akka Streams or we would know.

**Review by patriknw — APPROVED** [review]
LGTM, but I'm not sure if there was a concern with updating this?

**Review by johanandren — APPROVED** [review]

**Pull request merged**